frontend/src/pages/keygenPeerDiscovery/KeygenPeerDiscoveryView.tsx

```diff
@@ -0,0 +1,158 @@
+
+  useEffect(() => {
+    if (sessionID && serviceName) {
+      discoverService(serviceName);
+    }
+  }, [isRelay, sessionID, serviceName]);
+
+  const discoverService = async (name: string) => {
+    if (!isRelay) await AdvertiseMediator(name);
+    await postSession(isRelay, sessionID!, serviceName!);
+    checkForDevices(isRelay, sessionID!, setDevices);
+  };
+
+  const handleCanContinue = () => {
+    const minDevices = vaultType.split('/')[1];
+    switch (minDevices) {
+      case 'n':
+        return selectedDevices.length < 1;
+      case '3':
+        return selectedDevices.length == 2 ? false : true;
+      case '2':
+        return selectedDevices.length == 1 ? false : true;
+    }
+  };
+
+  const startKeygen = async () => {
+    if (handleCanContinue()) {
+      await startkeygen(isRelay, sessionID!, selectedDevices);
+    }
+  };
+
+  return (
+    <>
+      <div className="mx-auto w-full  text-white text-center">
+        <div>
+          {/* sample data */}
+          <KeygenQRCode data={qrData} serviceName={serviceName!} />
+        </div>
+        <div className="flex gap-10 justify-center mt-5">
+          <button
+            onClick={() => setIsRelay(true)}
+            className={`${
+              isRelay ? 'bg-switchBtn-secondary' : 'bg-switchBtn-primary'
+            } rounded-3xl flex items-center justify-center w-[150px] py-2 gap-2`}
+          >
+            <img src="/assets/images/icons/cellular.svg" alt="cellular" />{' '}
+            {t('internet')}
+          </button>
+          <button
+            onClick={() => setIsRelay(false)}
+            className={`${
+              !isRelay ? 'bg-switchBtn-secondary' : 'bg-switchBtn-primary'
+            } rounded-3xl flex items-center justify-center w-[150px] py-2 gap-2`}
+          >
+            <img src="/assets/images/icons/wifi.svg" alt="wifi" /> {t('local')}
+          </button>
+        </div>
+        {devices.length == 0 ? (
+          <>
+            <h3 className="mt-5 font-semibold">{t('looking_for_devices')}</h3>
+            <div className="w-[100px] h-auto mx-auto">
+              <Lottie animationData={loadingAnimation} loop={true} />
+            </div>
+          </>
+        ) : (
+          <SelectDevice
+            devices={devices}
+            selectedDevices={selectedDevices}
+            setSelectedDevices={setSelectedDevices}
+          />
+        )}
+
+        {isRelay ? (
+          <div>
+            <img
+              src="/assets/images/icons/cellular.svg"
+              alt="cellular"
+              className="mx-auto my-2"
+            />
+            {t('devices_on_same_internet')}
+          </div>
+        ) : (
+          <div>
+            <img
+              src="/assets/images/icons/wifi.svg"
+              alt="wifi"
+              className="mx-auto my-2"
+            />
+            {t('peer_discovery_hint_connect_to_same_wifi')}
+          </div>
+        )}
+        <button
+          disabled={handleCanContinue()}
```

Comment:
disabled = handleCanContinue
in startKeygen function, it checks if it is handleCanContinue
does it work? This means, keygen can't be started? Not sure how it worked in your testing.